tests/test_state_machines.py

```diff
@@ -103,6 +104,25 @@ def test_client_publish_qos0(
     assert packet.payload == "payload"
 
 
+@pytest.mark.parametrize("qos", [QoS.AT_MOST_ONCE, QoS.AT_LEAST_ONCE, QoS.EXACTLY_ONCE])
+def test_client_limit_qos(qos) -> None:
+    """Test that a limited QoS is processed in the client"""
+    client = MQTTClientStateMachine(client_id="client-X")
+    client.connect()
+    assert client.state is MQTTClientState.CONNECTING
+    _bytes = client.get_outbound_data()  # ignored
```

Comment:
Is this line even needed?
```suggestion
```

Reply:
Probably not in this case. I'll drop it.